lift subsystem

Mboesn opened the pull request.

Mboesn committed:
lift subsystem

nobody went over it doh

HilelAmiel commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -23,40 +26,46 @@
   private static final String kCustomAuto = "My Auto";
```
first of all we want the code to be clean from comments, please remove all comments that created automaticly when creating the robot project.

HilelAmiel commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -23,40 +26,46 @@
+   * This function is run when the robot is first started up and should be used
+   * for any initialization code.
    */
   @Override
```
also here

HilelAmiel commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -23,40 +26,46 @@
     SmartDashboard.putData("Auto choices", m_chooser);
+    lift = new Lift(new VictorSP(RobotMap.LIFT_LEFT_MOTOR), new VictorSP(RobotMap.LIFT_RIGHT_MOTOR),
+        new Encoder(RobotMap.LIFT_ENCODER_CHANNELA, RobotMap.LIFT_ENCODER_CHANNELB));
+
```
names: LIFT_ENCODER_CHANNEL_A, LIFT_ENCODER_CHANNEL_B

esh4 replied:
Not using encoders...

HilelAmiel commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -23,40 +26,46 @@
+   * This runs after the mode specific periodic functions, but before LiveWindow
+   * and SmartDashboard integrated updating.
    */
   @Override
```
commants...

HilelAmiel commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -23,40 +26,46 @@
+   * You can add additional auto modes by adding additional comparisons to the
+   * switch structure below with additional strings. If using the SendableChooser
+   * make sure to add them to the chooser code above as well.
    */
```
and commants...

HilelAmiel commented on src/main/java/frc/robot/RobotMap.java:
```diff
@@ -0,0 +1,20 @@
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
```
delete commants

HilelAmiel commented on src/main/java/frc/robot/RobotMap.java:
```diff
@@ -0,0 +1,20 @@
+    public static final int LIFT_RIGHT_MOTOR = 1;
+    public static final int LIFT_ENCODER_CHANNELA = 0;
+    public static final int LIFT_ENCODER_CHANNELB = 1;
+
```
i want you to make it more clear which value (like LIFT_LEFT_MOTOR) is port of PWM (motor controllers) and which values are ports of DIO (sensors).

HilelAmiel commented on src/main/java/frc/robot/Subsystems/Lift.java:
```diff
@@ -0,0 +1,42 @@
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
```
delete commants

HilelAmiel commented on src/main/java/frc/robot/Subsystems/Lift.java:
```diff
@@ -0,0 +1,42 @@
+/**
+ * Add your docs here.
+ */
+public class Lift extends Subsystem {
```
also here

HilelAmiel commented on src/main/java/frc/robot/Subsystems/Lift.java:
```diff
@@ -0,0 +1,42 @@
+  SpeedController liftMotorRight, liftMotorLeft;
+  Encoder encoder;
+  SpeedControllerGroup liftMotorGroup = new SpeedControllerGroup(this.liftMotorLeft, this.liftMotorRight);
+
```
names: liftRightMotor, liftLeftMotor, liftEncoder
and make the SpeedControllerGroup ( = new SpeedControllerGroup(this.liftMotorLeft, this.liftMotorRight); ) in the class constractor after the client write which liftRightMotor and liftLeftMotor he use in the main class -> Robot.java

HilelAmiel commented on src/main/java/frc/robot/Subsystems/Lift.java:
```diff
@@ -0,0 +1,42 @@
+
+  public double encoderTicks() {
+    return encoder.get();
+  }
```
we want to know what is the hight **(in meters)** the lift system got not in ticks, so make a math calculation (if you use parameters we dont know put then in robot map (yeah fine put them there) and make TODO there that tell us we need to masure them or find what are they when we will have robot...
also change the function name to getHight()

HilelAmiel commented on src/main/java/frc/robot/Subsystems/Lift.java:
```diff
@@ -0,0 +1,42 @@
+  }
+
+  public void setMotorSpeed(double power) {
+    liftMotorGroup.set(power);
```
name: double speed (not power)

HilelAmiel commented on src/main/java/frc/robot/Subsystems/Lift.java:
```diff
@@ -0,0 +1,42 @@
+  public void initDefaultCommand() {
+    // Set the default command for a subsystem here.
+    // setDefaultCommand(new MySpecialCommand());
+  }
```
delete commants

HilelAmiel reviewed (changes requested):
great code, pay attention to my commants i think you need to start rechange things from robot class.

Mboesn committed:
Fixed by code review

- comments
~ changed names
TODO change ticks to meters

Mboesn committed:
Math + Fix upon code review

+ RobotConstents
+ MathFunction
+ Pulse to meter
~ changed objects to private
+ EncoderReset

Mboesn committed:
Added comments

Mboesn committed:
comments

ya

Mboesn committed:
Fixed objects (NOT FINISHED)

~ encoder to potentoimeter
+ 2 microswitch
~ victor to SRX
(TODO finish caculations set potentiometer to caculation)

Mboesn committed:
Fixed values

~ fixed values for potentiometer

esh4 commented on src/main/java/frc/robot/RobotConstants.java:
```diff
@@ -0,0 +1,44 @@
+         * and not in turns of the tube.
+         */
+        public static final double LIFT_POTENTIOMETER_SCALE_FRACTOR = RobotDimensions.LIFT_TUBE_CIRCUMFERENCE;
+        /**
```
Factor or franction

esh4 commented on src/main/java/frc/robot/Subsystems/Lift.java:
```diff
@@ -0,0 +1,51 @@
+  /** This function returns the curent state of the potentiometer. */
+  public AnalogPotentiometer getPotentiometer() {
+    return this.potentiometer;
+  }
```
Returns a double

Mboesn committed:
Fixed based on review

~ to .get()
~ names

Mboesn committed:
Swicthed to TalonSRX

oh yeah yeah we gonna take over

esh4 commented on src/main/java/frc/robot/Subsystems/Lift.java:
```diff
@@ -0,0 +1,54 @@
+
+  /** This function returns the curent state of the potentiometer. */
+  public Double getPotentiometer() {
+    return this.potentiometer.get();
```
double

Mboesn committed:
Fixed based on code review

~ Double to double

esh4 committed:
Merge branch 'master' into Lift
